crypto/http/http_client.c

```diff
@@ -689,6 +687,11 @@ int OSSL_HTTP_REQ_CTX_nbio(OSSL_HTTP_REQ_CTX *rctx)
         rctx->state = OHS_ERROR;
         return 0;
 
+        /*
+         * TODO break up the state machine at this point (between write and read).
+         * Also factor out bulky code sections in aux functions. See issue #26665
```

Comment:
Instead of TODO, which means unfinished code not for merging to a master branch, please formulate this like: 

`The state machine could be broken up at this point and the bulky code sections factorized out.`

Also referring to issue numbers in code might not be a good idea as issues get closed for various reasons and this comment might survive longer time than expected.